compiler/rustc_mir_transform/src/uninhabited_enum_branching.rs

```diff
@@ -99,15 +124,33 @@ impl<'tcx> MirPass<'tcx> for UninhabitedEnumBranching {
             if let TerminatorKind::SwitchInt { targets, .. } =
                 &mut body.basic_blocks_mut()[bb].terminator_mut().kind
             {
-                let new_targets = SwitchTargets::new(
+                let mut new_targets = SwitchTargets::new(
                     targets.iter().filter(|(val, _)| allowed_variants.contains(val)),
                     targets.otherwise(),
                 );
 
-                *targets = new_targets;
+                if new_targets.iter().count() == allowed_variants.len() {
+                    if let Some(updated) = ensure_otherwise_unreachable(body, &new_targets) {
+                        let new_otherwise = body.basic_blocks_mut().push(updated);
+                        *new_targets.all_targets_mut().last_mut().unwrap() = new_otherwise;
+                        needs_cleanup = true;
+                    }
+                }
+
+                if let TerminatorKind::SwitchInt { targets, .. } =
+                    &mut body.basic_blocks_mut()[bb].terminator_mut().kind
+                {
+                    *targets = new_targets;
+                } else {
+                    unreachable!()
+                }
             } else {
                 unreachable!()
             }
         }
+
+        if needs_cleanup {
+            super::simplify::simplify_cfg(tcx, body);
+        }
```

Comment:
The current pipeline includes simplify cfg immediately after this pass, so I would omit cleanup here.

https://github.com/rust-lang/rust/blob/0c292c9667f1b202a9150d58bdd2e89e3e803996/compiler/rustc_mir_transform/src/lib.rs#L450-L451